### ISSUE TITLE
SSL Certificate pinning broken for Android 5+

### ISSUE THREAD

**Issue by suzukieng** [Countly/countly-sdk-android#58]
CertificateTrustManager.checkServerTrusted() expects authType to be RSA, but starting with Android 5+, 

```
        if (!(null != authType && authType.equalsIgnoreCase("RSA"))) {
            throw new CertificateException("PublicKeyManager: AuthType is not RSA");
        }
```

See comment in: http://developer.android.com/about/versions/android-5.0-changes.html#ssl

```
App is making wrong assumptions about cipher suites used to connect to server
For example, some apps contain a custom X509TrustManager that breaks because it expects the authType parameter to be RSA but encounters ECDHE_RSA or DHE_RSA.
```

Exception stack trace:

```
01-05 14:18:08.777 22583 22600 W Countly : javax.net.ssl.SSLHandshakeException: PublicKeyManager: AuthType is not RSA
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:328)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.http.SocketConnector.connectTls(SocketConnector.java:103)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.Connection.connect(Connection.java:143)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.Connection.connectAndSetOwner(Connection.java:185)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:128)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.http.HttpEngine.nextConnection(HttpEngine.java:341)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:330)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:248)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:433)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:114)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.connect(DelegatingHttpsURLConnection.java:89)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java)
01-05 14:18:08.777 22583 22600 W Countly :  at ly.count.android.sdk.ConnectionProcessor.run(ConnectionProcessor.java:175)
01-05 14:18:08.777 22583 22600 W Countly :  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
01-05 14:18:08.777 22583 22600 W Countly :  at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-05 14:18:08.777 22583 22600 W Countly :  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
01-05 14:18:08.777 22583 22600 W Countly :  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
01-05 14:18:08.777 22583 22600 W Countly :  at java.lang.Thread.run(Thread.java:818)
01-05 14:18:08.777 22583 22600 W Countly : Caused by: java.security.cert.CertificateException: PublicKeyManager: AuthType is not RSA
01-05 14:18:08.777 22583 22600 W Countly :  at ly.count.android.sdk.CertificateTrustManager.checkServerTrusted(CertificateTrustManager.java:55)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.org.conscrypt.Platform.checkServerTrusted(Platform.java:117)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.org.conscrypt.OpenSSLSocketImpl.verifyCertificateChain(OpenSSLSocketImpl.java:556)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
01-05 14:18:08.777 22583 22600 W Countly :  at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:324)
01-05 14:18:08.777 22583 22600 W Countly :  ... 17 more
```

**Comment by iartem** [comment]
Hi @suzukieng 
Thanks for your report. The fix will be part of next SDK version which is expected to be out this month.

**Comment by iartem** [comment]
Hi @suzukieng 
I'm sorry, but we moved release of new SDK to later date. 16.02 release won't have this issue fixed.

**Comment by iartem** [comment]
Fixed a while ago in #4da1354
